### PR TITLE
Pass `--raw` or `--raw-except=datalad` to addurl

### LINKED ISSUE #26: return addurl --raw as  --raw-except=datalad
will need not yet released git-annex as fixed in [10.20240129-42-g6b38d0c427](https://git.kitenet.net/index.cgi/git-annex.git/commit/?id=6b38d0c427a645d421ebf2d21111c0ca224dfef3)

### PR DESCRIPTION
**Pull request by jwodder**
Closes #26.

### CODE REVIEW

**Comment by jwodder** [comment]
The tests are currently failing because neurodebian's git-annex-standalone is still on version 10.20231227.

**Comment by yarikoptic** [comment]
to merge we need to ensure that recent enough git-annex is within that env:

```shell
dandi@drogon:/mnt/backup/dandi/dandisets$ source ~/.bashrc-miniconda

(base) dandi@drogon:/mnt/backup/dandi/dandisets$ conda activate dandisets-2

(dandisets-2) dandi@drogon:/mnt/backup/dandi/dandisets$ git-annex version 
git-annex version: 10.20230126-g36f5557
...
(dandisets-2) dandi@drogon:/mnt/backup/dandi/dandisets$ which git-annex
/home/dandi/miniconda3/envs/dandisets-2/bin/git-annex
```

but most recent annex is not on conda-forge due to problems building that ghc etc
- https://github.com/conda-forge/git-annex-feedstock/pull/160

any clues on how to progress that one forward? (alternative would be to remove conda installation of git-annex and rely on neurodebian one, but it would be slower since standalone build)

**Comment by jwodder** [comment]
@yarikoptic Does drogon have a recent enough version of `git-annex` that this can be merged now?

**Comment by jwodder** [comment]
@yarikoptic Ping.

**Comment by yarikoptic** [comment]
yes, we have git-annex upgraded systemwide and I removed its installation from conda-forge's environment:

```
dandi@drogon:~$ source ~/.bashrc-miniconda; conda activate dandisets-2
(dandisets-2) dandi@drogon:~$ git annex version --raw
10.20240430-1~ndall+1(dandisets-2) dandi@drogon:~$ which git-annex
/usr/bin/git-annex
(dandisets-2) dandi@drogon:~$ conda deactivate
(base) dandi@drogon:~$ git annex version --raw
10.20240430-1~ndall+1(base) 
```

File: README.md [hunk 1]
```diff
@@ -17,7 +17,7 @@ Before running `backups2datalad`, the following setup must be performed:
   required.
 
 - [git-annex](https://git-annex.branchable.com) must be installed.  At least
-  version 10.20230126 is required, though you should endeavor to obtain the
+  version 10.20240227 is required, though you should endeavor to obtain the
```
Comment (yarikoptic) [inline README.md:20]: would you be so kind to add explicit check for git-annex version, e.g. 

```shell
❯ python -c 'from datalad.support.external_versions import external_versions as externals; print(externals["cmd:annex"] >= "10.20240227")'
False
❯ git annex version
git-annex version: 10.20240129
```

so we do not need to deal with aftermath of potential running with outdated ones and crashing only later on, requiring cleaning etc

Comment (jwodder) [reply README.md:20]: What does that code snippet do if git-annex isn't installed?  Would `externals["cmd:annex"]` be `None` or absent?

Comment (yarikoptic) [reply README.md:20]:
```shell
❯ python -c 'from datalad.support.external_versions import external_versions as externals; print(repr(externals["cmd:annex"]))'
LooseVersion ('10.20240129')
❯ sudo dpkg --purge --force-depends git-annex
...
❯ python -c 'from datalad.support.external_versions import external_versions as externals; print(repr(externals["cmd:annex"]))'
None
```

Comment (jwodder) [reply README.md:20]: Done: c81b62b

**Comment by jwodder** [comment]
Tests are failing because the Archive hasn't updated to the new dandischema: https://github.com/dandi/dandi-archive/issues/1975

**Comment by jwodder** [comment]
@yarikoptic Tests are passing now.  Can this finally be merged?

File: src/backups2datalad/util.py [hunk 2]
```diff
@@ -281,3 +282,19 @@ def yaml_dump(data: Any) -> str:
 
 class UnexpectedChangeError(Exception):
     pass
+
+
+def check_git_annex_version() -> None:
+    # Call this function at the start of subcommand functions rather than in
+    # `main()` so that it doesn't run if a user does `backups2datalad
+    # subcommand --help`
```
Comment (yarikoptic) [inline src/backups2datalad/util.py:290]: I bet there should be a way for a common to all subcommands CLI context handling outside of --help, or may be in the top level context handling to do it is there is no --version or --help

But ok for now to get us going

**Comment by yarikoptic** [comment]
Thank you! I left a comment but let's proceed

**Pull request merged**